pkg/generators/clientgen/gen.go

```diff
@@ -37,15 +37,20 @@ import (
 	"github.com/kcp-dev/code-generator/pkg/flag"
 	"github.com/kcp-dev/code-generator/pkg/internal"
 	"github.com/kcp-dev/code-generator/pkg/util"
+	genutil "k8s.io/code-generator/cmd/client-gen/generators/util"
 )
 
 var (
-	// RuleDefinition is a marker for defining rules
-	ruleDefinition = markers.Must(markers.MakeDefinition("genclient", markers.DescribesType, placeholder{}))
-	// nonNamespacedMarker checks if resource is namespaced or clusterscoped
-	nonNamespacedMarker = markers.Must(markers.MakeDefinition("genclient:nonNamespaced", markers.DescribesType, placeholder{}))
-	// noStatusMarker checks if status is to scaffolded
-	noStatusMarker = markers.Must(markers.MakeDefinition("+genclient:noStatus", markers.DescribesType, placeholder{}))
+	genclientMarker     = markers.Must(markers.MakeDefinition("genclient", markers.DescribesType, genclient{}))
+	nonNamespacedMarker = markers.Must(markers.MakeDefinition("genclient:nonNamespaced", markers.DescribesType, struct{}{}))
+	noStatusMarker      = markers.Must(markers.MakeDefinition("genclient:noStatus", markers.DescribesType, struct{}{}))
+	noVerbsMarker       = markers.Must(markers.MakeDefinition("genclient:noVerbs", markers.DescribesType, struct{}{}))
+	readOnlyMarker      = markers.Must(markers.MakeDefinition("genclient:readonly", markers.DescribesType, struct{}{}))
+
+	// These two markers are defined separately since they accept a list which is comma separated. In controller-tools, comma
```

Comment:
What does "defined separately" mean?

Reply:
In controller-tool's terms the following format is how the marker is defined: `<makername>:<parameter>=<values>`. As a consumer of this library, I need to register (what `MakeDefinition` does here) just the marker name once. But there are 2 edge cases, because of which we are defining the marker formats separately.
1. The parameter does not take any value. That's why `genclient:noVerbs`, `genclient:readonly` are registered as separate markers.
2. The value accepts a list. Controller-tool's parsing logic identifies a list only if it is separated by ";", whereas genclinet formats have comma separated values. This confuses the parser, and screws up the logic. To work around this, we define this as separate markers. 
Just as an example - this is the reason why `genclient:Method` marker is not defined explicitly, but others are.

Reply:
So by "separately" you mean that we are passsing `markers.RawArguments("")`?

Reply:
Not exactly, I meant that we are registering it to markers list explicitly using `MakeDefinition` (ie) this:
https://github.com/kcp-dev/code-generator/blob/d5edbfdfa54f819b08c5285b1ffe1e7c9bc6bf54/pkg/generators/clientgen/gen.go#L52-L53

The `method` marker (ie) `genclient:method` doesn't need that kind explicit registration because it's considered as a sub-type of `genclient` marker. `markers.RawArguments("")` is actually just parsing whatever value is being passed to the respective marker (say `genclient:onlyVerb`) and returning to us.

Reply:
Ah, I see - comparing _everything_ here (from `nonNamespacedMarker` down) to ones that exist but are not even listed. I see. I was trying to tell the difference between e.g. these two under the comment and the ones above them.